// gatsby/dist/redux/reducers/index.js
"use strict";

var _nodes = require("./nodes");

var _pages = require("./pages");

var _redirects = require("./redirects");

var _schema = require("./schema");

var _staticQueryComponents = require("./static-query-components");

var _status = require("./status");

var _webpack = require("./webpack");

var _pageData = require("./page-data");

var _themes = require("./themes");

var _webpackCompilationHash = require("./webpack-compilation-hash");

var _reducer = require("gatsby-cli/lib/reporter/redux/reducer");

var _lastAction = require("./last-action");

var _jobsv = require("./jobsv2");

var _pageDataStats = require("./page-data-stats");

var _components = require("./components");

var _componentDataDependencies = require("./component-data-dependencies");

var _nodesTouched = require("./nodes-touched");

var _babelrc = require("./babelrc");

var _jobs = require("./jobs");

const nodesByType = require(`./nodes-by-type`);

/**
 * @property exports.nodesTouched Set<string>
 */
module.exports = {
  program: require(`./program`),
  nodes: _nodes.nodeReducer,
  nodesByType: nodesByType,
  resolvedNodesCache: require(`./resolved-nodes`),
  nodesTouched: _nodesTouched.nodesTouchedReducer,
  lastAction: _lastAction.lastAction,
  flattenedPlugins: require(`./flattened-plugins`),
  config: require(`./config`),
  schema: _schema.schemaReducer,
  pages: _pages.pagesReducer,
  status: _status.statusReducer,
  componentDataDependencies: _componentDataDependencies.componentDataDependenciesReducer,
  components: _components.componentsReducer,
  staticQueryComponents: _staticQueryComponents.staticQueryComponentsReducer,
  jobs: _jobs.jobsReducer,
  jobsV2: _jobsv.jobsV2Reducer,
  webpack: _webpack.webpackReducer,
  webpackCompilationHash: _webpackCompilationHash.webpackCompilationHashReducer,
  redirects: _redirects.redirectsReducer,
  babelrc: _babelrc.babelrcReducer,
  schemaCustomization: require(`./schema-customization`),
  themes: _themes.themesReducer,
  logs: _reducer.reducer,
  inferenceMetadata: require(`./inference-metadata`),
  pageDataStats: _pageDataStats.pageDataStatsReducer,
  pageData: _pageData.pageDataReducer
};
